ci/scripts/determine-deployment-type.ts

```diff
@@ -0,0 +1,12 @@
+const semver = require('semver');
+
+const version = process.argv[2];
+
+let deploymentType;
+if (semver.prerelease(version)) {
+    deploymentType = 'canary';
+} else {
+    deploymentType = 'stable';
+}
```

Comment:
maybe some sort of validation wouldn't hurt here? I  mean, in case process.argv[2] is whatever bogus (foo, undefined). you get deployment type 'stable'. 

maybe semver.valid could do the thing